thorlcr/thorutil/thormisc.cpp

```diff
@@ -332,9 +332,19 @@ class DECL_EXCEPTION CThorException : public CSimpleInterface, implements IThorE
             }
             if (slave)
             {
-                str.appendf("SLAVE #%d [", slave);
-                queryClusterGroup().queryNode(slave).endpoint().getUrlStr(str);
-                str.append("]: ");
+                if (((int)slave) < 0)
+                {
+                    unsigned node = (unsigned)-slave;
+                    str.appendf("NODE #%d [", node);
+                    queryNodeGroup().queryNode(node).endpoint().getUrlStr(str);
+                    str.append("]: ");
+                }
+                else
+                {
+                    str.appendf("SLAVE #%d [", slave);
+                    queryClusterGroup().queryNode(slave).endpoint().getUrlStr(str);
```

Comment:
@jakesmith slave was previously serialized 0 based, now 1 based.  Are the calls to queryNode() correct?  Also it ok, that slave number in tracing used to be 0 based, now 1 based?

Reply:
previously it was serialized 0 based, but the deserialization code used to make it 1 based in the exception [ by calling e->setSlave(slave+1) ].

So it remains 1 based here and needs to be, 0 of node group is master.
Tracing was/remains 1 based as before.

Reply:
You're correct, I missed that.